File: backend/src/main/java/com/rafael/clients/application/service/ClientService.java
package com.rafael.clients.application.service;

import java.util.List;
import java.util.UUID;

import org.springframework.stereotype.Service;

import com.rafael.clients.domain.model.Client;
import com.rafael.clients.domain.service.ClientDomainService;

@Service
public class ClientService {

    private final ClientDomainService clientDomainService;

    public ClientService(ClientDomainService clientDomainService) {
        this.clientDomainService = clientDomainService;
    }

    public Client createClient(Client client) {
        return clientDomainService.createClient(client);
    }

    public Client getClientById(UUID id) {
        return clientDomainService.findById(id);
    }

    public List<Client> getClients() {
        return clientDomainService.findAll();
    }

    public Client updateClient(UUID id, Client entity) {
        return clientDomainService.updateClient(id, entity);
    }

    public void deleteClient(UUID id) {
        clientDomainService.deleteClient(id);
    }
}

File: frontend/src/app/modules/clients/services/client.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Client } from '../models/client.model';

@Injectable({
  providedIn: 'root'
})
export class ClientService {
  private readonly apiUrl = 'http://localhost:8080/api/clients';

  constructor(private http: HttpClient) { }

  getClients(page: number = 1, limit: number = 10, search?: string): Observable<Client[]> {
    return this.http.get<Client[]>(this.apiUrl);
  }

  getClientById(id: string): Observable<Client> {
    return this.http.get<Client>(`${this.apiUrl}/${id}`);
  }

  createClient(client: Client): Observable<Client> {
    console.log(client);
    return this.http.post<Client>(this.apiUrl, client);
  }

  updateClient(id: string, Client: Partial<Client>): Observable<Client> {
    return this.http.put<Client>(`${this.apiUrl}/${id}`, Client);
  }

  deleteClient(id: string): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}